spec/app/domain/authentication/authn_oidc/authenticator_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnOidc::Authenticator do
+  def non_equal_id_token_subject
+    "other-sub"
+  end
+
+  def valid_user_info_subject
+    "some-sub"
+  end
+
+  def valid_user_info
+    double('user_info',
+            sub: valid_user_info_subject,
+            preferred_username: "non_nil_value"
+          )
+  end
+
+  def no_profile_scope_user_info
+    double('user_info',
+            sub: valid_user_info_subject,
+            preferred_username: nil
+          )
+  end
+
+  let (:authenticator_instance) do
+    Authentication::AuthnOidc::Authenticator.new(env:[])
+  end
+
+  it "validates user_info with valid value" do
+    subject = authenticator_instance
+    expect { subject.send(:validate_user_info, valid_user_info, valid_user_info_subject) }.to_not raise_error
+  end
```

Comment:
We should never unit test private methods.  Only test the public behavior of the object.

Reply:
I agree. At this point this is the only thing we have a UT for so I removed the file